compiler/damlc/lib/DA/Cli/Damlc/Command/MultiIde/Forwarding.hs

```diff
@@ -0,0 +1,174 @@
+  -> LSP.Message m
+  -> Forwarding m
+getMessageForwardingBehaviour meth params =
+  case meth of
+    LSP.SInitialize -> error "Forwarding of Initialize must be handled elsewhere."
+    LSP.SInitialized -> ignore
+    -- send to all then const reply
+    LSP.SShutdown -> ForwardRequest params $ AllRequest (assumeSuccessCombiner @m $ const LSP.Empty)
+    LSP.SExit -> ForwardNotification params AllNotification
+    LSP.SWorkspaceDidChangeWorkspaceFolders -> ForwardNotification params AllNotification
+    LSP.SWorkspaceDidChangeConfiguration -> ForwardNotification params AllNotification
+    LSP.SWorkspaceDidChangeWatchedFiles -> ForwardNotification params AllNotification
+    LSP.STextDocumentDidOpen -> ForwardNotification params $ forwardingBehaviourFromParamsWithTextDocument params
+    LSP.STextDocumentDidChange -> ForwardNotification params $ forwardingBehaviourFromParamsWithTextDocument params
+    LSP.STextDocumentWillSave -> ForwardNotification params $ forwardingBehaviourFromParamsWithTextDocument params
+    LSP.STextDocumentWillSaveWaitUntil -> ForwardRequest params $ forwardingBehaviourFromParamsWithTextDocument params
+    LSP.STextDocumentDidSave -> ForwardNotification params $ forwardingBehaviourFromParamsWithTextDocument params
+    LSP.STextDocumentDidClose -> ForwardNotification params $ forwardingBehaviourFromParamsWithTextDocument params
+    LSP.STextDocumentCompletion -> ForwardRequest params $ forwardingBehaviourFromParamsWithTextDocument params
+    LSP.STextDocumentHover -> ForwardRequest params $ forwardingBehaviourFromParamsWithTextDocument params
+    LSP.STextDocumentSignatureHelp -> ForwardRequest params $ forwardingBehaviourFromParamsWithTextDocument params
+    LSP.STextDocumentDeclaration -> ForwardRequest params $ forwardingBehaviourFromParamsWithTextDocument params
+    LSP.STextDocumentDefinition -> error "Forwarding of STextDocumentDefinition must be handled elsewhere."
+    LSP.STextDocumentDocumentSymbol -> ForwardRequest params $ forwardingBehaviourFromParamsWithTextDocument params
+    LSP.STextDocumentCodeAction -> ForwardRequest params $ forwardingBehaviourFromParamsWithTextDocument params
+    LSP.STextDocumentCodeLens -> ForwardRequest params $ forwardingBehaviourFromParamsWithTextDocument params
+    LSP.STextDocumentDocumentLink -> ForwardRequest params $ forwardingBehaviourFromParamsWithTextDocument params
+    LSP.STextDocumentColorPresentation -> ForwardRequest params $ forwardingBehaviourFromParamsWithTextDocument params
+    LSP.STextDocumentOnTypeFormatting -> ForwardRequest params $ forwardingBehaviourFromParamsWithTextDocument params
+    
+    LSP.SCustomMethod "daml/keepAlive" ->
+      case params of
+        LSP.ReqMess LSP.RequestMessage {_id, _method, _params} -> ExplicitHandler $ \sendClient _ ->
+          sendClient $ LSP.FromServerRsp _method $ LSP.ResponseMessage "2.0" (Just _id) (Right Aeson.Null)
+        _ -> error "Got unpexpected daml/keepAlive response from client"
```

Comment:
maybe just ignore with a BIG WARNING popup notification? A general method would be good

Reply:
Added notifications :)